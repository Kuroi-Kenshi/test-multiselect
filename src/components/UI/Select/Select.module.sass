.wrapper
  margin: 10px
.select
  width: 200px
  height: 30px
  border-radius: 5px
  border: 2px solid rgba(34, 60, 80, 0.2)
  outline: none
