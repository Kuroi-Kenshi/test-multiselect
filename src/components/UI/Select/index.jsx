import React from 'react';
import { eventGenerator } from '../../../utils/eventGenerator';
import s from './Select.module.sass'

const Select = ({ name, onChange, value, options = [], showTitle = false, loading }) => {
  return (
    <div className={s.wrapper}>
      <select
        name={name}
        onChange={onChange}
        value={value}
        className={s.select}
        data-event={eventGenerator("Click on Select", '')}
        disabled={loading}
      >
        {showTitle ? <option value="" className={s.title}> {loading ? "Загрузка данных" : "Выберите категорию"}</option> : false}
        {options.map(item => {
          return (
            <option key={item.id} value={item.id} data-event={eventGenerator("Parent category selected", item)}>{item.name}</option>
          )
        })
        }
      </select>
    </div>
  );
};

export default React.memo(Select);
