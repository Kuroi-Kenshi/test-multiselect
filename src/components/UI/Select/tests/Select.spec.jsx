import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Select from '../index';

const onChangeSelect = () => { }
const category = '';
const categories = [
  {
    "id": 1,
    "name": "Cars",
    "flags": ""
  },
  {
    "id": 2,
    "name": "Motorcycles",
    "flags": ""
  },
  {
    "id": 3,
    "name": "Bicycles",
    "flags": ""
  },
  {
    "id": 4,
    "name": "Scooters",
    "flags": ""
  },
  {
    "id": 5,
    "name": "Airplanes",
    "flags": ""
  }
]

const setUp = (props) => shallow(<Select {...props} />);

// beforeEach(() => {
//   component = setUp({
//     name: "categories",
//     onChange: onChangeSelect,
//     value: category,
//     options: categories,
//   });
// });


describe("should render Select component", () => {

  test("matches snapshot1", () => {
    const wrapper = setUp({
      name: "categories",
      onChange: onChangeSelect,
      value: category,
      options: categories,
      showTitle: true
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  test("render without options", () => {
    const wrapper = setUp({
      name: "categories",
      onChange: onChangeSelect,
      value: category,
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})