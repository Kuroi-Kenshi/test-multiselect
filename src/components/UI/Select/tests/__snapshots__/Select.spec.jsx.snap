// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Select component matches snapshot1 1`] = `
<div
  className="wrapper"
>
  <select
    className="select"
    data-event="{ \\"id\\": \\"1636955733180\\", \\"eventName\\": \\"Click on Select\\", \\"eventValue\\": \\"\\"}"
    name="categories"
    onChange={[Function]}
    value=""
  >
    <option
      className="title"
      value=""
    >
       
      Выберите категорию
    </option>
    <option
      data-event="{ \\"id\\": \\"1636955733180\\", \\"eventName\\": \\"Parent category selected\\", \\"eventValue\\": {\\"id\\":1,\\"name\\":\\"Cars\\",\\"flags\\":\\"\\"}}"
      key="1"
      value={1}
    >
      Cars
    </option>
    <option
      data-event="{ \\"id\\": \\"1636955733180\\", \\"eventName\\": \\"Parent category selected\\", \\"eventValue\\": {\\"id\\":2,\\"name\\":\\"Motorcycles\\",\\"flags\\":\\"\\"}}"
      key="2"
      value={2}
    >
      Motorcycles
    </option>
    <option
      data-event="{ \\"id\\": \\"1636955733180\\", \\"eventName\\": \\"Parent category selected\\", \\"eventValue\\": {\\"id\\":3,\\"name\\":\\"Bicycles\\",\\"flags\\":\\"\\"}}"
      key="3"
      value={3}
    >
      Bicycles
    </option>
    <option
      data-event="{ \\"id\\": \\"1636955733180\\", \\"eventName\\": \\"Parent category selected\\", \\"eventValue\\": {\\"id\\":4,\\"name\\":\\"Scooters\\",\\"flags\\":\\"\\"}}"
      key="4"
      value={4}
    >
      Scooters
    </option>
    <option
      data-event="{ \\"id\\": \\"1636955733180\\", \\"eventName\\": \\"Parent category selected\\", \\"eventValue\\": {\\"id\\":5,\\"name\\":\\"Airplanes\\",\\"flags\\":\\"\\"}}"
      key="5"
      value={5}
    >
      Airplanes
    </option>
  </select>
</div>
`;

exports[`should render Select component render without options 1`] = `
<div
  className="wrapper"
>
  <select
    className="select"
    data-event="{ \\"id\\": \\"1636955733190\\", \\"eventName\\": \\"Click on Select\\", \\"eventValue\\": \\"\\"}"
    name="categories"
    onChange={[Function]}
    value=""
  />
</div>
`;
